### PR TITLE
ref(pyupgrade): class fixers wholesale on src/

### PR DESCRIPTION
**Pull request by joshuarli**
Some classes of fixes are obvious enough that we just made the decision to apply to the rest of src. See https://github.com/getsentry/sentry/pull/23631.

Again as I mentioned in that PR, some trivial yield fixers are included in the mix.

### CODE REVIEW

**Comment by joshuarli** [comment]
Awwwww shit.

```
  File "/home/runner/work/sentry/sentry/src/sentry/utils/services.py", line 102, in <lambda>
    context[key] = (lambda f: lambda *a, **k: getattr(self, f)(*a, **k))(key)
  File "/home/runner/work/sentry/sentry/src/sentry/utils/services.py", line 70, in __getattr__
    self._setup()
  File "/home/runner/work/sentry/sentry/src/sentry/utils/services.py", line 94, in _setup
    instance = backend(**self._options)
  File "/home/runner/work/sentry/sentry/src/sentry/tsdb/redissnuba.py", line 126, in __init__
    super().__init__(**options)
TypeError: super(type, obj): obj must be an instance or subtype of type
Error: Process completed with exit code 1.
```

File: src/bitfield/models.py [hunk 1]
```diff
@@ -20,8 +20,7 @@ def __repr__(self):
         return repr(self._flags)
 
     def __iter__(self):
-        for flag in self._flags:
-            yield flag
+        yield from self._flags
```
Comment (wedamija) [inline src/bitfield/models.py:23]: Woah

**Review by wedamija — APPROVED** [review]

File: src/bitfield/models.py (hunk 1 above)
Comment (joshuarli) [reply src/bitfield/models.py:23]: Good feature.

Comment (joshuarli) [reply src/bitfield/models.py:23]: Thanks Python 3.3.

**Review by evanpurkhiser — APPROVED** [review]
Didn't look at it all but looks good

**Pull request merged**